src/github_runner_manager/openstack_cloud/openstack_cloud.py

```diff
@@ -106,38 +106,71 @@ def __init__(self, server: OpenstackServer, prefix: str):
         self.status = server.status
 
 
+P = ParamSpec("P")
+T = TypeVar("T")
+
+
+def _catch_openstack_errors(func: Callable[P, T]) -> Callable[P, T]:
+    """Decorate a function to wrap OpenStack exceptions in a custom exception.
+
+    Args:
+        func: The function to decorate.
+
+    Returns:
+        The decorated function.
+    """
+
+    @functools.wraps(func)
+    def exception_handling_wrapper(*args: P.args, **kwargs: P.kwargs) -> T:
+        """Wrap the function with exception handling.
+
+        Args:
+            args: The positional arguments.
+            kwargs: The keyword arguments.
+
+        Raises:
+            OpenStackError: If any OpenStack exception is caught.
+
+        Returns:
+            The return value of the decorated function.
+        """
+        try:
+            return func(*args, **kwargs)
+        except (
+            openstack.exceptions.SDKException,
+            keystoneauth1.exceptions.ClientException,
+        ) as exc:
+            logger.error("OpenStack API call failure")
+            raise OpenStackError("Failed OpenStack API call") from exc
+
+    return exception_handling_wrapper
+
+
 @contextmanager
-@retry(tries=2, delay=5, local_logger=logger)
```

Comment:
This did not work anyway, the context manager ensures that the codepath in the retry fct was never reached.